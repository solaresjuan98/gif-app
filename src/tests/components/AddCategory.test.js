import React from 'react';
import { shallow } from "enzyme";
import { AddCategory } from '../../compontents/AddCategory'


describe('test on <AddCategory /> component', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    //
    beforeEach(() => {
        jest.clearAllMocks();//clear simulations
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should to show correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should change textbox', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {
            target: {
                value: value
            }
        });

        expect(wrapper.find('p').text().trim()).toBe(value);

    });


    test('should not post the info with submit', () => {

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        });

        expect(setCategories).not.toHaveBeenCalled();
    });


    test('should call setCategories and clear textbox', () => {

        const value = 'Hola';

        wrapper.find('input').simulate('change', {
            target: { value }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        });
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    });


});
