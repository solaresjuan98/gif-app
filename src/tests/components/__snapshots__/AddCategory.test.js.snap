// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <AddCategory /> component should to show correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      className="form-control"
      id="inputDefault"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
